Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.IntegrationTests/ApiGatewayEmulatorProcessTests.cs

```diff
@@ -290,15 +290,107 @@ public async Task TestLambdaWithNullEndpoint()
         }
     }
 
-    private async Task<HttpResponseMessage> TestEndpoint(string routeName, int apiGatewayPort, string httpMethod = "POST")
+    [Theory]
+    [InlineData(ApiGatewayEmulatorMode.Rest)]
+    [InlineData(ApiGatewayEmulatorMode.HttpV1)]
+    public async Task TestLambdaWithLargeRequestPayload_RestAndV1(ApiGatewayEmulatorMode mode)
+    {
+        var (lambdaPort, apiGatewayPort) = await GetFreePorts();
+        _cancellationTokenSource = new CancellationTokenSource();
+        _cancellationTokenSource.CancelAfter(TimeSpan.FromSeconds(120));
+        var consoleError = Console.Error;
+        try
+        {
+            Console.SetError(TextWriter.Null);
+            await StartTestToolProcessAsync(mode, "largerequestfunction", lambdaPort, apiGatewayPort, _cancellationTokenSource);
+            await WaitForGatewayHealthCheck(apiGatewayPort);
+
+            var handler = (APIGatewayProxyRequest request, ILambdaContext context) =>
+            {
+                var env = Environment.GetEnvironmentVariable("AWS_LAMBDA_RUNTIME_API");
+                testOutputHelper.WriteLine($"TestLambdaWithLargeRequestPayload {mode}: {env}");
+                return new APIGatewayProxyResponse
+                {
+                    StatusCode = 200,
+                    Body = request.Body.Length.ToString()
+                };
+            };
+
+            Environment.SetEnvironmentVariable("AWS_LAMBDA_RUNTIME_API", $"localhost:{lambdaPort}/largerequestfunction");
```

Comment:
Coordinate with @philasmar PR so that you can set the API via a command line instead of environment variable. https://github.com/aws/aws-lambda-dotnet/pull/1992